bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqData.java

```diff
@@ -82,16 +85,14 @@ public GarageDoorData getMyqData() {
 	 *            Attempt number when it recursively calls itself
 	 */
 	private String getGarageStatus(int attemps) {
-		if (this.sercurityTokin == null && Login()) {
+		if (sercurityTokin == null && !Login()) {
```

Comment:
@digitaldan I tried to make it one line instead of two embedded if statement that I first wrote. I didn't want to call Login every time the polling executes and over work the API. Now that you changed the Login method to throws it could be simplified to this instead with the return null removed:
```
if(sercurityTokin == null){
	Login();
}
```
My goal was to call login if the sercurityTokin is null for when the binding starts up. The sercurityTokin also expires after 15 minutes and I wanted to add a timer and check the elapsed time since the last login but I didn't know how to do this in java. I just added the recursion so it would retry to login if it failed to get the device status on the first attempt. This is how I wrote it in my C# demo app I was testing the API with:
```
if (string.IsNullOrEmpty(sercurityTokin) || nowTime.Subtract(loginTime) >= TimeSpan.FromMinutes(15))
{
      Login(userName, password);
}
```

I plan on adding this in once I figure out how to do this in java.